ngraph/frontend/paddlepaddle/src/op/stack.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <ngraph/opsets/opset8.hpp>
+#include <node_context.hpp>
+
+namespace ngraph {
+namespace frontend {
+namespace pdpd {
+namespace op {
+NamedOutputs stack(const NodeContext& node) {
+    auto datas = node.get_ng_inputs("X");
+    int axis = 0;
+    if (node.has_attribute<int32_t>("axis"))
+        axis = node.get_attribute<int32_t>("axis");
+
+    auto data_shape = datas[0].get_partial_shape();
+    auto data_type = datas[0].get_element_type();
+    int64_t data_rank = datas[0].get_partial_shape().rank().get_length();
+    if (axis < 0)
+        axis = axis + data_rank + 1;
+    OutputVector node_datas_reshape;
+
+    auto axis_const = std::make_shared<opset8::Constant>(element::i64, Shape{}, axis);
+    for (const auto& data : datas) {
+        PDPD_OP_VALIDATION_CHECK(node, data.get_partial_shape().rank().is_static(), "stack rank must be static");
+        PDPD_OP_VALIDATION_CHECK(node,
+                                 data_shape == data.get_partial_shape(),
+                                 "stack input tensor must the same shape!");
+        PDPD_OP_VALIDATION_CHECK(node,
+                                 data_type == data.get_element_type(),
+                                 "stack input tensor must the same data types!");
+
+        node_datas_reshape.push_back(std::make_shared<opset8::Unsqueeze>(data, axis_const));
```

Comment:
Unsqueeze supports negative axis and may also remove the check between line 27~33 to support dynamic shape, Concat should do the check itself.

Reply:
removed the static shape check, while kept the original shape and element type check of this op.